tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/JMSXPathSelectorTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.tests.integration.amqp;
+
+import javax.jms.Connection;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import org.apache.activemq.artemis.api.core.QueueConfiguration;
+import org.apache.activemq.artemis.api.core.RoutingType;
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.impl.AddressInfo;
+import org.apache.activemq.artemis.core.settings.impl.AddressSettings;
+import org.junit.Test;
+
+public class JMSXPathSelectorTest extends JMSClientTestSupport {
+
+   private static final String NORMAL_QUEUE_NAME = "NORMAL";
+
+   private ConnectionSupplier CoreConnection = () -> createCoreConnection();
+   private ConnectionSupplier OpenWireConnection = () -> createOpenWireConnection();
+
+   @Override
+   protected String getConfiguredProtocols() {
+      return "OPENWIRE,CORE";
```

Comment:
Why not amqp too? I dont get why its excluded here

Reply:
The body for text messages sent by STOMP, OpenWire, and core JMS are all nullable `SimpleString` so it was super easy to pull that out in the filter implementation. I just didn't have the desire to fish around in the AMQP code to figure out how to do it although nothing is stopping anybody from implementing this in the future.

Reply:
Of the top of my head....

    if (message instanceof AMQPMessage) {
        final Section body = ((AMQPMessage) message).getBody();
        if (body instanceof AmqpValue && ((AmqpValue) body).getValue() instanceof String) {
           return (String) ((AmqpValue) body).getValue();
        }
    }

Reply:
@tabish121, thanks for the tip.

@michaelandrepearce, I updated the PR with support for AMQP with the caveat that the Qpid JMS client chokes on the `XPATH` selector syntax.

Reply:
One could probably use the "jms.validateSelector" URI option [1] (or related setter on the connection factory) to disable the clients selector validation and let the illegal-for-JMS selector string value through unchecked.

[1] http://qpid.apache.org/releases/qpid-jms-0.56.0/docs/index.html#jms-configuration-options